File: services/backend/nutrition_service/src/main/kotlin/com/example/nutrition_service/persistence/pojos/RecipeModel.kt
package com.example.nutrition_service.persistence.pojos

data class RecipeModel(
    val nutrients: NutrientModel,
    val timeTotal: TimeTotalModel,
    val name: String,
    val description: String,
    val keywords: String,
    val categories: List<CategoryModel>,
    val images: List<ImageModel>?,
    val instructions: List<InstructionModel>?
)

File: services/frontend/keep-fit/src/types/models/nutrition/RecipeModel.ts
import TimeTotalModel from "./TimeTotalModel";
import CategoryModel from "./CategoryModel";
import ImagesModel from "./ImagesModel";
import InstructionModel from "./InstructionModel";
import IngredientModel from "./IngredientModel";
import NutrientModel from "./NutrientModel";

type RecipeModel = {
    id: number;
    timeTotal: TimeTotalModel;
    name: string;
    description: string;
    keywords: string;
    categories: CategoryModel[];
    images: ImagesModel[];
    instructions: InstructionModel[];
    ingredients: IngredientModel[];
    nutrients: NutrientModel;
};

export default RecipeModel;
